app/services/variant_units/unit_prices.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module VariantUnits
+  class UnitPrices
+    def initialize(variant)
+      @variant = variant
+      @product = variant.product
+    end
+
+    def unit_price_denominator
+      return @variant.unit_value if @product&.variant_unit == "items"
+
+      case unit_price_unit
+      when "lb"
+        @variant.unit_value / 453.6
+      when "kg"
+        @variant.unit_value / 1000
+      else # Liters
+        @variant.unit_value
+      end
+    end
+
+    def unit_price_unit
+      return "lb" if VariantUnits::WeightsAndMeasures.new(@variant).
```

Comment:
You shouldn't need `VariantUnits` here provided both these classes live inside that namespace. I'd pretty remove this namespace altogether if you ask me. I don't see what's the actual value it provides.